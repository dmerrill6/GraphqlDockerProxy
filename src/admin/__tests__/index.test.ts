import { getAdminSchema } from '../index'

jest.mock('graphql-tools', () => {
  return {
    makeExecutableSchema: (obj) => {
      return obj
    },
    mergeSchemas: (obj) => {
      return obj
    },
  }
})

jest.mock('../k8sSchema', () => {
  return {
    typeDefs: `k8sSchema Admin`,
    resolvers: {
      mock: true,
    },
  }
})

jest.mock('../generalSchema', () => {
  return {
    typeDefs: `generalSchema Admin`,
    resolvers: {
      mock: true,
    },
  }
})

describe('tests getAdminSchema', () => {
  const runtime = ['docker', 'kubernetesWatch', 'dockerWatch', 'kubernetes']

  runtime.map((one) => {
    it('tests by runtime: ' + one, () => {
      process.env.qglProxyRuntime = one
      expect(getAdminSchema()).toMatchSnapshot()
    })
  })

})
