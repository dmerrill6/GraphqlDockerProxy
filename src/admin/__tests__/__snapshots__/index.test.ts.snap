// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests getAdminSchema tests by runtime: docker 1`] = `
Object {
  "schemas": Array [
    Object {
      "resolvers": Object {
        "mock": true,
      },
      "typeDefs": "generalSchema Admin",
    },
  ],
}
`;

exports[`tests getAdminSchema tests by runtime: dockerWatch 1`] = `
Object {
  "schemas": Array [
    Object {
      "resolvers": Object {
        "mock": true,
      },
      "typeDefs": "generalSchema Admin",
    },
  ],
}
`;

exports[`tests getAdminSchema tests by runtime: kubernetes 1`] = `
Object {
  "schemas": Array [
    Object {
      "resolvers": Object {
        "mock": true,
      },
      "typeDefs": "k8sSchema Admin",
    },
    Object {
      "resolvers": Object {
        "mock": true,
      },
      "typeDefs": "generalSchema Admin",
    },
  ],
}
`;

exports[`tests getAdminSchema tests by runtime: kubernetesWatch 1`] = `
Object {
  "schemas": Array [
    Object {
      "resolvers": Object {
        "mock": true,
      },
      "typeDefs": "k8sSchema Admin",
    },
    Object {
      "resolvers": Object {
        "mock": true,
      },
      "typeDefs": "generalSchema Admin",
    },
  ],
}
`;
